build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -702,6 +703,38 @@ message NodeProperty {
     string value = 2;
 }
 
+// Node properties for [FileNodes][build.bazel.remote.execution.v2.FileNode],
+// [DirectoryNodes][build.bazel.remote.execution.v2.DirectoryNode], and
+// [SymlinkNodes][build.bazel.remote.execution.v2.SymlinkNode]. The server is
+// responsible for specifying the properties that it accepts.
+//
+message NodeProperties {
+  // A list of string-based
+  // [NodeProperties][build.bazel.remote.execution.v2.NodeProperty].
+  repeated NodeProperty properties = 1;
+
+  // The file's last modification timestamp.
+  google.protobuf.Timestamp mtime = 2;
+
+  // The UNIX file mode, e.g., 0755.
+  google.protobuf.UInt32Value unix_mode = 3;
+}
+
+// A set of node property keys including string-based and field-based variants.
+message NodePropertyKeys {
```

Comment:
Now that the values of field-based properties are always embedded messages (to distinguish between the absence of a value and the default value), `NodePropertyKeys` is no longer needed in the directory trees and thus, space efficiency for specifying the set of node property keys is less of a concern.

We could consider dropping `NodePropertyKeys` and keep using `repeated string` fields for capabilities and `output_node_properties` even for field-based properties. This would match the approach taken by the protobuf `FieldMask` concept.

The only downside I see is that mixing field names and names of string-based node properties might be confusing. If field names always start with a lowercase letter and string-based node property names always start with an uppercase letter, there shouldn't be any real conflicts, though.

Any thoughts on this?

Reply:
I think I would stick with the simpler approach (repeated string) for now.

Reply:
Ok, I've switched back to the simpler `repeated string` approach, dropping `NodePropertyKeys`.